allensdk/brain_observatory/behavior/behavior_project_api/behavior_ophys_analysis_query_utils.py

```diff
@@ -0,0 +1,309 @@
+from pymongo import MongoClient
+import yaml
+import pandas as pd
+import numpy as np
+import json
+import os
+import glob
+import traceback
+import datetime
+
+import time
+from tqdm import tqdm as progress
+
+from IPython.core.display import display, HTML
+display(HTML("<style>.container { width:100% !important; }</style>"))
+
+from allensdk.brain_observatory.behavior import behavior_project_cache as bpc
+import allensdk.brain_observatory.behavior.cell_metrics as cm
+
+class Database(object):
+    '''
+    utilities for connecting to MongoDB databases (mouseseeks or visual_behavior_data)
+
+    parameter:
+      database: defines database to connect to. Can be 'mouseseeks' or 'visual_behavior_data'
+    '''
+
+    def __init__(self, database_server='visual_behavior_data', ):
+        # get database ip/port info from a text file on the network (maybe not a good idea to commit it)
+        db_info_filepath = '//allen/programs/braintv/workgroups/nc-ophys/visual_behavior/mongo_db_info.yml'
+        with open(db_info_filepath, 'r') as stream:
+            db_info = yaml.safe_load(stream)
+
+        # connect to the client
+        ip = db_info[database_server]['ip']
+        port = db_info[database_server]['port']
+        self.client = MongoClient('mongodb://{}:{}'.format(ip, port))
+
+        # set each database as an attribute of the class (but not admin) and as an entry in a dictionary
+        # this will provide flexibility in how the databases are called
+        self.database = {}
+        self.database_names = []
+        databases = [db for db in self.client.list_database_names() if db != 'admin']
+        for db in databases:
+            self.database_names.append(db)
+            self.database[db] = self.client[db]
+            setattr(self, db, self.client[db])
+        # make subscriptable
+        self._db_names = {db: self.client[db] for db in databases}
+
+    def __getitem__(self, item):
+        # this allows databases to be accessed by name
+        return self._db_names[item]
+
+    def query(self, database, collection, query={}, return_as='dataframe'):
+        '''
+        Run a query on a collection in the database.
+        The query should be formated as set of key/value pairs
+        Sending an empty query will return the entire collection
+        '''
+
+        return pd.DataFrame(list(self.database[database][collection].find(query)))
+
+    def close(self):
+        '''
+        close connection to client
+        '''
+        self.client.close()
+        
+def is_int(n):
+    return isinstance(n, (int, np.integer))
+
+def is_float(n):
+    return isinstance(n, (float, np.float))
+
+def clean_and_timestamp(entry):
+    entry = {k: (int(v) if is_int(v) else v) for k, v in entry.items()}
+    entry = {k: (float(v) if is_float(v) else v) for k, v in entry.items()}
+    entry.update({'entry_time_utc':str(datetime.datetime.utcnow())})
+    return entry
+
+def write_to_manifest_collection(manifest,overwrite = False):
```

Comment:
maybe run this file through a formatter like [black](https://pypi.org/project/black/)

Reply:
used autopep8